docs_theme/main.html

```diff
@@ -0,0 +1,57 @@
+{% extends "base.html" %}
+
+{% block scripts %}
+{{ super() }}
+<script>
+  fetch('{{ 'versions.txt'|url }}')
+    .then(function(res) {if (res.ok) return res.text();})
+    .then(function(text) {return text.trim().split("\n");})
```

Comment:
I'm pretty sure this function could be called with undefined and throw an error. The above `then` only returns the text if the res is OK, a 404 will not throw an error and thus it will pass undefined to this function.

Reply:
I added a codepath for res not OK.